hyperspy/misc/utils.py

```diff
@@ -1091,6 +1091,104 @@ def func(*args):
+                               nav_indexes=None,
+                               output_signal_size=None,
+                               block_info=None,
+                               **kwargs):
+    """
+    Function for processing the function blockwise...
+    This gets passed to map_blocks so that the function
+    only gets applied to the signal axes.
+
+    Parameters:
+    ------------
+    data: np.array
+        The data for one chunk
+    *args: np.array
+        Any signal the is iterated alongside the data
+    function: function
+        The function to applied to the signal axis
+    nav_shape: tuple
+        The shape of the navigation axes
+    output_signal_shape: tuple
+        The shape of the output signal.  For a ragged signal
+        this is equal to 1.
+    dtype: np.dtype
+        The datatype for the output array. For preallocating.
+        For a ragged array this is np.object.
+    iterating_kwargs: list str
+        The keys for anything that is being mapped
+        alongside the data.
+    **kwargs: dict
+        Any additional key value pairs to be used by the function
+        (Note that these are the constants that are applied.)
+
+    """
+    # Both of these values need to be passed in
+    dtype = block_info[None]["dtype"]
+    chunk_nav_shape = tuple([data.shape[i] for i in sorted(nav_indexes)])
+    output_shape = chunk_nav_shape + tuple(output_signal_size)
+    # Pre-allocating the output array
+    output_array = np.empty(output_shape, dtype=dtype)
+    if len(args) == 0:
+        # There aren't any BaseSignals for iterating
+        for nav_index in np.ndindex(chunk_nav_shape):
+            islice = np.s_[nav_index]
+            output_array[islice] = function(data[islice],
+                                            **kwargs)
+    else:
+        # There are BaseSignals which iterate alongside the data
+        for index in np.ndindex(chunk_nav_shape):
+            islice = np.s_[index]
+            iter_dict = {a[0]: a[1][islice].squeeze() for a in args}
+            output_array[islice] = function(data[islice],
+                                            **iter_dict,
+                                            **kwargs)
+    try:
+        output_array = output_array.squeeze(-1)
+    except ValueError:
+        pass
+    return output_array
+
+
+def rechunk_signal(signal,
+                   key,
+                   nav_chunks,
+                   test=False,
+                   ):
+    test_kwarg = {}
+    if test:
+        test_ind = (0,) * len(signal.axes_manager.navigation_axes)
+        if signal.axes_manager.signal_shape == ():
+            test_kwarg[key] = signal.inav[test_ind].data[0]
+        else:
+            test_kwarg[key] = signal.inav[test_ind].data
+    signal = signal.as_lazy()
+    new_chunks = nav_chunks + (*signal.axes_manager.signal_shape,)
+    signal.data = signal.data.rechunk(new_chunks)
+    rechunked_signal = (key, signal.data)
+    return rechunked_signal, test_kwarg
```

Comment:
This seems to be a left over from previous version. If so, it should be removed.